test/three/basic_test.go

```diff
@@ -0,0 +1,29 @@
+package three
+
+import (
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	util "github.com/rancher/terraform-provider-rancher2/test"
+)
+
+func TestThreeBasic(t *testing.T) {
+	t.Parallel()
+	config := util.NewTestConfig(t, "use-cases/three")
+
+	defer config.Teardown(t)
```

Comment:
IIUC `Teardown` will delete the directory where the kubeconfig is stored, right? If so then we should get the logs first, and then tear it all down.

```suggestion
	defer func() {
	    config.GetErrorLogs(t)
	    config.Teardown(t)
	}()	
```

Reply:
I had to look this up, defer happens in LIFO order, so deferring the logs after the teardown will get the logs first.

Reply:
https://go.dev/tour/flowcontrol/13

Reply:
TIL! In the future it may be better to defer an anonymous function so that the execution order is more clear, but as long as this works as expected I think its fine for now